cmd/bundle/exec.go

```diff
@@ -0,0 +1,152 @@
+
+The current working directory of the provided command will be set to the root
+of the bundle.
+
+Example usage:
+1. databricks bundle exec -- echo "hello, world"
+2. databricks bundle exec -- /bin/bash -c "echo hello"
+3. databricks bundle exec -- uv run pytest`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.ArgsLenAtDash() != 0 {
+				return fmt.Errorf("Please add a '--' separator. Usage: 'databricks bundle exec -- %s'", strings.Join(args, " "))
+			}
+
+			// Load the bundle configuration to get the authentication credentials.
+			b, diags := root.MustConfigureBundle(cmd)
+			if diags.HasError() {
+				return diags.Error()
+			}
+
+			childCmd := exec.Command(args[0], args[1:]...)
+
+			env := auth.ProcessEnv(root.ConfigUsed(cmd.Context()))
+
+			// If user has specified a target, pass it to the child command. DABs
+			// defines a "default" target which is a placeholder for when no target is defined.
+			// If that's the case, i.e. no targets are defined, then do not pass the target.
+			//
+			// This is only useful for when the Databricks CLI is the child command.
+			if b.Config.Bundle.Target != mutator.DefaultTargetPlaceholder {
+				env = append(env, "DATABRICKS_BUNDLE_TARGET="+b.Config.Bundle.Target)
+			}
+
+			// If the bundle has a profile configured, explicitly pass it to the child command.
+			//
+			// This is only useful for when the Databricks CLI is the child command,
+			// since if we do not explicitly pass the profile, the CLI will use the
+			// auth configured in the bundle YAML configuration (if any).
+			if b.Config.Workspace.Profile != "" {
+				env = append(env, "DATABRICKS_CONFIG_PROFILE="+b.Config.Workspace.Profile)
+			}
+
+			childCmd.Env = env
+
+			// Execute all scripts from the bundle root directory. This behavior can
+			// be surprising in isolation, but we do it to keep the behavior consistent
+			// for both these cases:
+			// 1. One shot commands like `databricks bundle exec -- echo hello`
+			// 2. (upcoming) Scripts that are defined in the scripts section of the DAB.
+			//
+			// TODO(shreyas): Add a DATABRICKS_BUNDLE_INITIAL_CWD environment variable
+			// that users can read to figure out the original CWD. I'll do that when
+			// adding support for the scripts section.
+			childCmd.Dir = b.BundleRootPath
+
+			stdout, err := childCmd.StdoutPipe()
+			if err != nil {
+				return fmt.Errorf("creating stdout pipe failed: %w", err)
+			}
+
+			stderr, err := childCmd.StderrPipe()
+			if err != nil {
+				return fmt.Errorf("creating stderr pipe failed: %w", err)
+			}
+
+			// Start the child command.
+			err = childCmd.Start()
+			if err != nil {
+				return fmt.Errorf("starting %q failed: %w", strings.Join(args, " "), err)
+			}
+
+			var wg sync.WaitGroup
+			wg.Add(2)
+
+			var stdoutErr error
+			go func() {
+				defer wg.Done()
+
+				scanner := bufio.NewScanner(stdout)
+				for scanner.Scan() {
+					_, err = fmt.Fprintln(cmd.OutOrStdout(), scanner.Text())
```

Comment:
Why capture & re-print the output here? Why not just connect file descriptors directly?

Actually, why run a subprocess instead of execv()?

Reply:
> Why not just connect file descriptors directly?

That was my first attempt, that does not flush all the output. The main CLI process exits before the file descriptors are completely flushed.

> why run a subprocess instead of execv()

I don't think the standard library directly supports `execv`. Could not find a ref by a quick google or perplexity check. In any case either approach works as long as the end sematics meet our requirements. There's still the open question regarding whether exit codes should be propogated.